packages/adblocker/src/engine/engine.ts

```diff
@@ -79,6 +79,12 @@ export interface BlockingResponse {
         contentType: string;
         dataUrl: string;
       };
+  substitude:
```

Comment:
? Nit. I'm also not sure how to understand the name `substitute`. Can we make it more specific? Maybe `mutatedUrl` or `modifiedUrl` or `rewrittenUrl`?
```suggestion
  substitute:
```

Reply:
After thinking a bit, I feel having a term of "action" as a top level property name looks promising. Therefore, I took `urlRewrite` or maybe `urlMutation` for a top level property name.

```
export interface BlockingResponse {
  match: boolean;
  redirect:
    | undefined
    | {
        body: string;
        contentType: string;
        dataUrl: string;
      };
  urlRewrite:
    | undefined
    | {
        rewrittenUrl: string;
        filters: NetworkFilter[];
      };
```

Do you think this structure is clear?

Reply:
This was updated in 22cd3d71075ce5279916e6b88f56fbbde657beee, please check.